retrievalmarket/impl/client_test.go

```diff
@@ -225,6 +225,7 @@ func TestClient_FindProviders(t *testing.T) {
 // retrieval deal for the same payload CID with the same peer as an existing
 // active deal
 func TestClient_DuplicateRetrieve(t *testing.T) {
+	t.Skip("flaky test")
```

Comment:
Did it become more flaky in this PR?
1. Yes -- then let's figure that out
2. No -- I would not throw in a skip as part of a feature PR -- i'd just make an issue and if we can't tackle it now, let's make a seperate PR to add a skip

Reply:
2. Issue at https://github.com/filecoin-project/go-fil-markets/issues/683.